qa/build.gradle

```diff
@@ -39,8 +39,15 @@ task pullBwcPlugin {
     }
 
     doLast {
+        ext{
+            if (knn_bwc_version == "2.4.0") {
+                srcUrl = "https://ci.opensearch.org/ci/dbc/distribution-build-opensearch/${knn_bwc_version}/latest/linux/x64/tar/dist/opensearch/opensearch-${knn_bwc_version}-linux-x64.tar.gz"
```

Comment:
When we can rollback this logic to a uniform url for all versions? Do you know if build team will move the artifact once 2.4.0 is released, or this url will still work?

Reply:
I think it will still work because when I have tried to download the 2.3.0 version using same url, it worked https://ci.opensearch.org/ci/dbc/distribution-build-opensearch/2.3.0/latest/linux/x64/tar/dist/opensearch/opensearch-2.3.0-linux-x64.tar.gz. But, IMO we should always test against the released version. So, we can comment out this piece of code after 2.4.0 is released

Reply:
Right, that's my concern that we use the actual released artifact and not the latest, as build will work it's on our good intensions to change that URL. We need some sort of action item for ourselves to make sure that change is not abandoned. Could you please create an issue against knn 2.4 to revert url related logic once core 2.4 got finalized for release?

Reply:
Makes sense. Will create an issue pointing to this piece of code which needs to be removed in the future after this PR is merged

Reply:
I think this will change once we have released version of 2.4 . This change cannot be reverted till we have a maven release for 2.4 or any other version. I feel this change will be there forever in the 2.4 branch, as we should not make the change in 2.4 branch once it is released

Reply:
We may need to revert this in "main", for 2.4 branch - maybe too, it depends on when core team release the final artifact for 2.4 so the url looks similar to other releases.

Reply:
We don't need to make this change in 2.4 branch because we are using this logic and fetching latest artifacts for bwc versions. In 2.4 branch we will only use released versions to upgrade to 2.4.0 version. So, it is not a problem even if the code is not reverted.